From: longlong yan <yanlonglong@kylinos.cn>
Subject: selftests/mm/mseal_test: Fix mmap return value check in test_seal_zero_address
Date: Tue, 7 Jul 2026 11:17:45 +0800

The return value for a failed mmap function is MAP_FAILED.  According to
the definition of FAIL_TEST_IF FALSE,ptr must be determined to be distinct
from MAP_FAILED

Link: https://lore.kernel.org/20260707031745.1039-1-yanlonglong@kylinos.cn
Fixes: 4926c7a52de7 ("selftest mm/mseal memory sealing")
Signed-off-by: longlong yan <yanlonglong@kylinos.cn>
Acked-by: Mike Rapoport (Microsoft) <rppt@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/mseal_test.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/testing/selftests/mm/mseal_test.c~selftests-mm-mseal_test-fix-mmap-return-value-check-in-test_seal_zero_address
+++ a/tools/testing/selftests/mm/mseal_test.c
@@ -482,7 +482,7 @@ static void test_seal_zero_address(void)
 	/* use mmap to change protection. */
 	ptr = mmap(0, size, PROT_NONE,
 		   MAP_ANONYMOUS | MAP_PRIVATE | MAP_FIXED, -1, 0);
-	FAIL_TEST_IF_FALSE(ptr == 0);
+	FAIL_TEST_IF_FALSE(ptr != (void *)-1);
 
 	size = get_vma_size(ptr, &prot);
 	FAIL_TEST_IF_FALSE(size == 4 * page_size);
_
